### DISCUSSION TITLE
Should plugin-instructions' on_page_change be triggered once upon trial start?

### DISCUSSION THREAD

**Discussion by Shaobin-Jiang** [jspsych/jsPsych#3350] [General]
Let's say we want the back nav button to be hidden on the page. We would have to do something like this:

```javascript
let trial = {
    type: jsPsychInstructions,
    pages: ['<p>Page 1</p>', '<p>Page 2</p>', '<p>Page 3</p>', '<p>Page 4</p>'],
    show_clickable_nav: true,
    on_page_change: page_change,
    on_load: function () {
        page_change(0);
    },
};

function page_change(page) {
    if (page === 0) {
        document.querySelector('#jspsych-instructions-back').style.visibility = 'hidden';
    }
}
```

This is no good, having to call `page_change` twice. If we allow `on_page_change` to run once upon trial start, the problem should be solved.

Of course, there might be occasions when users do not want this behavior. So I am proposing that we change `on_page_change` to a function that receives two arguments: the current page and the page from which it came. In this way, the user has a way of knowing whether the current run of `on_page_change` is triggered by trial start or really by a page change.

**Comment by jodeleeuw** [comment]
That sounds good to me! 👍
